### ISSUE TITLE
query_factory getBindVarValue float not returned

### ISSUE THREAD

**Issue by mc12345678** [zencart/zencart#1824]
I was working on some user data entry verification and had mistakenly thought that by submitting the entered data through: `$result = $db->bindVars($sqlStatement, ':field:', $newValue, 'float');` that the $newValue would be forced/verified to be of data type float (cast to float); however, upon both action and further review found I was using the function incorrectly.  The method that processes this request, `getBindVarValue` of the mysql query_factory class, primarily addresses having an emptish value, otherwise it returns whatever was fed to it.

As far as direct impact on the existing ZC software, I do not really see where use of the above would become a problem as (using github to search the files) it seems that the operation is used only in a reporting tool.

In my situation, I was testing to see how the entry of a non-numeric character would be treated such as entering `10a` instead of a value of `10`.  The value of `10a` was permitted through as it was and when attempting to store that value (expected to be a float) into the database, I was provided a mysql error indicating that the field `10a` did not exist instead of storing the value 10.

Was wondering, am I using the function with the wrong mindset (thinking that it would "clean" the value) and/or should the feature be updated to return a float cast value instead of just the value itself?

A proposed change may be:
````php
      case 'float':
        return (!zen_not_null($value) || $value=='' || $value == 0) ? 0 : $value;
````
changed to:
````php
      case 'float':
        return (!zen_not_null($value) || $value=='' || $value == 0) ? 0 : (float)$value;
````

or is there a reason to keep the conversion as it is and use an alternate method such as added to ZC 1.5.6, and/or should a php function be the route to go before using the above replacement process? ZC 1.5.6 in some areas (for backwards compatibility) as well as in a general function file added a new function: 

````php
  function convertToFloat($input = 0) {
    if ($input === null) {
      return 0;
    }
    $val = preg_replace('/[^0-9,\.\-]/', '', $input);
    // do a non-strict compare here:
    if ($val == 0) {
      return 0;
    }
    return (float)$val;
  }
````

**Comment by drbyte** [comment]
It could probably be safely changed to:

`return empty($value) ? 0 : float($value);`

**Comment by drbyte** [comment]
Closed by PR #1839
